[박지석] sprint3

commicat2 opened the pull request:
## 요구사항
# 기본 요구사항

## Spring 프로젝트 초기화

- [x] Spring Initializr를 통해 zip 파일을 다운로드하세요.
    - [x] 빌드 시스템은 Gradle - Groovy를 사용합니다.
    - [x] 언어는 Java 17를 사용합니다.
    - [x] Spring Boot의 버전은 3.4.0입니다.
    - [x] GroupId는 com.sprint.mission입니다.
    - [x] ArtifactId와 Name은 discodeit입니다.
    - [x] packaging 형식은 Jar입니다
    - [x] Dependency를 추가합니다.
        - [x] Lombok
        - [x] Spring Web
- [x] zip 파일을 압축해제하고 원래 진행 중이던 프로젝트에 붙여넣기하세요. 일부 파일은 덮어쓰기할 수 있습니다.
- [x] application.properties 파일을 yaml 형식으로 변경하세요.
- [x] DiscodeitApplication의 main 메서드를 실행하고 로그를 확인해보세요.

## Bean 선언 및 테스트

- [x] File*Repository 구현체를 Repository 인터페이스의 Bean으로 등록하세요.
- [x] Basic*Service 구현체를 Service 인터페이스의 Bean으로 등록하세요.
- [x] JavaApplication에서 테스트했던 코드를 DiscodeitApplication에서 테스트해보세요.
    - [x] JavaApplication 의 main 메소드를 제외한 모든 메소드를 DiscodeitApplication클래스로 복사하세요.
    - [x] JavaApplication의 main 메소드에서 Service를 초기화하는 코드를 Spring Context를 활용하여 대체하세요.
    ```java
    // JavaApplication
    public static void main(String[] args) {
    // 레포지토리 초기화
    // ...
    // 서비스 초기화
    UserService userService = new BasicUserService(userRepository);
    ChannelService channelService = new BasicChannelService(channelRepository);
    MessageService messageService = new BasicMessageService(messageRepository, channelRepository, userRepository);
    
        // ...
    
    }
    
    // DiscodeitApplication
    public static void main(String[] args) {
    ConfigurableApplicationContext context = SpringApplication.run(DiscodeitApplication.class, args);
    // 서비스 초기화
    // context에서 Bean을 조회하여 각 서비스 구현체 할당 코드 작성하세요.
    UserService userService;
    ChannelService channelService;
    MessageService messageService;
    
        // ...
        
    }
    ```
    - [x]  JavaApplication의 main 메소드의 셋업, 테스트 부분의 코드를 DiscodeitApplication클래스로 복사하세요.
    ```java
    public static void main(String[] args) {
    // ...
    // 셋업
    User jpaUser = setupUser(userService);
    Channel jpaChannel = setupChannel(channelService);
    // 테스트
    messageCreateTest(messageService, jpaChannel, jpaUser);
    }
    ```

## Spring 핵심 개념 이해하기

- [x] JavaApplication과 DiscodeitApplication에서 Service를 초기화하는 방식의 차이에 대해 다음의 키워드를 중심으로 정리해보세요.
    - IoC Container
    - Dependency Injection
    - Bean

> 이 내용은 PR에 첨부해주세요.

## Lombok 적용

- [x] 도메인 모델의 getter 메소드를 @Getter로 대체해보세요.
- [x] Basic*Service의 생성자를 @RequiredArgsConstructor로 대체해보세요.

## 비즈니스 로직 고도화

- [x]  다음의 기능 요구 사항을 구현하세요.

# 추가 기능 요구사항

## 시간 타입 변경하기

- [x] 시간을 다루는 필드의 타입은 Instant로 통일합니다.
    - 기존에 사용하던 Long보다 가독성이 뛰어나며, 시간대(Time Zone) 변환과 정밀한 시간 연산이 가능해 확장성이 높습니다.

## 새로운 도메인 추가하기

- 도메인 모델 간 참조 관계를 참고하세요.
  ![](readme1.png)

- [x] 공통: 앞서 정의한 도메인 모델과 동일하게 공통 필드(id, createdAt, updatedAt)를 포함합니다.
- [x] ReadStatus
    - 사용자가 채널 별 마지막으로 메시지를 읽은 시간을 표현하는 도메인 모델입니다. 사용자별 각 채널에 읽지 않은 메시지를 확인하기 위해 활용합니다.
- [x] UserStatus
    - 사용자 별 마지막으로 확인된 접속 시간을 표현하는 도메인 모델입니다. 사용자의 온라인 상태를 확인하기 위해 활용합니다.
    - [x] 마지막 접속 시간을 기준으로 현재 로그인한 유저로 판단할 수 있는 메소드를 정의하세요.
        - 마지막 접속 시간이 현재 시간으로부터 5분 이내이면 현재 접속 중인 유저로 간주합니다.
- [x] BinaryContent
    - 이미지, 파일 등 바이너리 데이터를 표현하는 도메인 모델입니다. 사용자의 프로필 이미지, 메시지에 첨부된 파일을 저장하기 위해 활용합니다.
    - [x] 수정 불가능한 도메인 모델로 간주합니다. 따라서 updatedAt 필드는 정의하지 않습니다.
    - [x] User, Message 도메인 모델과의 의존 관계 방향성을 잘 고려하여 id 참조 필드를 추가하세요.
- [x]  각 도메인 모델 별 레포지토리 인터페이스를 선언하세요.
- 레포지토리 구현체(File, JCF)는 아직 구현하지 마세요. 이어지는 서비스 고도화 요구사항에 따라 레포지토리 인터페이스에 메소드가 추가될 수 있어요.

## DTO 활용하기

[DTO란?](https://www.codeit.kr/topics/sprint-sb-mission-3/lessons/11891)

## UserService 고도화

- 고도화
    - create
        - [x] 선택적으로 프로필 이미지를 같이 등록할 수 있습니다.
        - [x] DTO를 활용해 파라미터를 그룹화합니다.
            - 유저를 등록하기 위해 필요한 파라미터, 프로필 이미지를 등록하기 위해 필요한 파라미터 등
        - [x] username과 email은 다른 유저와 같으면 안됩니다.
        - [x] UserStatus를 같이 생성합니다.
    - find, findAll
        - DTO를 활용하여:
            - [x] 사용자의 온라인 상태 정보를 같이 포함하세요.
            - [x] 패스워드 정보는 제외하세요.
    - update
        - [x] 선택적으로 프로필 이미지를 대체할 수 있습니다.
        - [x] DTO를 활용해 파라미터를 그룹화합니다.
            - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터
    - delete
        - [x] 관련된 도메인도 같이 삭제합니다.
            - BinaryContent(프로필), UserStatus
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme2.png)

# AuthService 구현

- login
    - [x] username, password과 일치하는 유저가 있는지 확인합니다.
        - [x] 일치하는 유저가 있는 경우: 유저 정보 반환
        - [x] 일치하는 유저가 없는 경우: 예외 발생
    - [x] DTO를 활용해 파라미터를 그룹화합니다.
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme3.png)

## ChannelService 고도화

- 고도화
    - create
        - PRIVATE 채널과 PUBLIC 채널을 생성하는 메소드를 분리합니다.
        - [ ] 분리된 각각의 메소드를 DTO를 활용해 파라미터를 그룹화합니다.
        - PRIVATE 채널을 생성할 때:
            - [ ] 채널에 참여하는 User의 정보를 받아 User 별 ReadStatus 정보를 생성합니다.
            - [ ] name과 description 속성은 생략합니다.
        - PUBLIC 채널을 생성할 때에는 기존 로직을 유지합니다.
    - find
        - DTO를 활용하여:
            - [ ] 해당 채널의 가장 최근 메시지의 시간 정보를 포함합니다.
            - [ ] PRIVATE 채널인 경우 참여한 User의 id 정보를 포함합니다.
    - findAll
        - DTO를 활용하여:
            - [ ] 해당 채널의 가장 최근 메시지의 시간 정보를 포함합니다.
            - [ ] PRIVATE 채널인 경우 참여한 User의 id 정보를 포함합니다.
        - [ ] 특정 User가 볼 수 있는 Channel 목록을 조회하도록 조회 조건을 추가하고, 메소드 명을 변경합니다. findAllByUserId
        - [ ] PUBLIC 채널 목록은 전체 조회합니다.
        - [ ] PRIVATE 채널은 조회한 User가 참여한 채널만 조회합니다.
    - update
        - [ ] DTO를 활용해 파라미터를 그룹화합니다.
            - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터
        - [ ] PRIVATE 채널은 수정할 수 없습니다.
    - delete
        - [ ] 관련된 도메인도 같이 삭제합니다.
            - Message, ReadStatus
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme4.png)

## MessageService 고도화

- 고도화
    - create
        - [ ] 선택적으로 여러 개의 첨부파일을 같이 등록할 수 있습니다.
        - [ ] DTO를 활용해 파라미터를 그룹화합니다.
    - findAll
        - [ ] 특정 Channel의 Message 목록을 조회하도록 조회 조건을 추가하고, 메소드 명을 변경합니다.
          `findallByChannelId`
    - update
        - [ ] DTO를 활용해 파라미터를 그룹화합니다.
            - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터
    - delete
        - [ ] 관련된 도메인도 같이 삭제합니다.
            - 첨부파일(BinaryContent)
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme5.png)

## ReadStatusService 구현

- create
    - [ ] DTO를 활용해 파라미터를 그룹화합니다.
    - [ ] 관련된 Channel이나 User가 존재하지 않으면 예외를 발생시킵니다.
    - [ ] 같은 Channel과 User와 관련된 객체가 이미 존재하면 예외를 발생시킵니다.
- find
    - [ ] id로 조회합니다.
- findAllByUserId
    - [ ] userId를 조건으로 조회합니다.
- update
    - [ ] DTO를 활용해 파라미터를 그룹화합니다.
        - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터
- delete
    - [ ] id로 삭제합니다.
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme6.png)

## UserStatusService 고도화

- create
    - [x] DTO를 활용해 파라미터를 그룹화합니다.
    - [x] 관련된 User가 존재하지 않으면 예외를 발생시킵니다.
    - [x] 같은 User와 관련된 객체가 이미 존재하면 예외를 발생시킵니다.
- find
    - [x] id로 조회합니다.
- findAll
    - [x] 모든 객체를 조회합니다.
- update
    - [x] DTO를 활용해 파라미터를 그룹화합니다.
        - 수정 대상 객체의 id 파라미터, 수정할 값 파라미터
- updateByUserId
    - [x] userId 로 특정 User의 객체를 업데이트합니다.
- delete
    - [x] id로 삭제합니다.
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme7.png)

## BinaryContentService 구현

- create
    - [x] DTO를 활용해 파라미터를 그룹화합니다.
- find
    - [x] id로 조회합니다.
- findAllByIdIn
    - [x] id 목록으로 조회합니다.
- delete
    - [x] id로 삭제합니다.
- 의존성
    - 같은 레이어 간 의존성 주입은 순환 참조 방지를 위해 지양합니다. 다른 Service 대신 필요한 Repository 의존성을 주입해보세요.
      ![](readme8.png)

## 새로운 도메인 Repository 구현체 구현

- [x]  지금까지 인터페이스로 설계한 각각의 Repository를 JCF, File로 각각 구현하세요.
  ![](readme9.png)

## 멘토에게
- 바빠서 JavaApplication에 있는 메서드들을 모두 DiscodeitApplication으로 옮기진 못했습니다.
- [x] JavaApplication과 DiscodeitApplication에서 Service를 초기화하는 방식의 차이에 대해 다음의 키워드를 중심으로 정리해보세요.
    - IoC Container
      - JavaApplication: IoC 컨테이너가 없습니다. 객체를 new로 직접 만들고 순서·수명도 직접 관리합니다.
      - DiscodeitApplication: 애플리케이션 시작 시 ApplicationContext(IoC 컨테이너)가 올라가고, 컴포넌트 스캔으로 클래스들을 빈으로 등록·관리합니다. 객체 생성 순서, 라이프사이클(초기화/소멸), 설정값 바인딩 등을 컨테이너가 맡습니다.
    - Dependency Injection
      - JavaApplication: 의존성 주입이 자동화되지 않고 조립을 직접 합니다.
      - DiscodeitApplication: DI가 자동입니다. @Service, @RequiredArgsConstructor로 서비스를 선언하면 컨테이너가 타입/프로필에 맞는 빈을 찾아 넣습니다. @Profile("test"|"dev"|"prod")로 구현을 바꿔 끼우거나, 설정으로 교체가 가능합니다. 트랜잭션, AOP 프록시도 DI 과정에서 자연스럽게 얹힙니다.
    - Bean
      - JavaApplication: “빈” 개념이 없습니다. 그냥 일반 객체입니다. 싱글톤/스코프/라이프사이클 관리도 직접 해야 합니다.
      - DiscodeitApplication: @Service, @Repository, @Component, @Configuration+@Bean 등으로 빈으로 등록됩니다.
        - 스코프: 기본 싱글톤, 필요 시 @Scope 변경.
        - 프로필: @Profile("test") 같은 애노테이션으로 환경별 빈 활성화.
        - 설정 주입: @ConfigurationProperties/@Value로 외부 설정 값 주입.
        - 라이프사이클: @PostConstruct/@PreDestroy, SmartLifecycle 등 컨테이너가 호출.

commicat2 committed:
feat: Spring 프로젝트 초기화

commicat2 committed:
docs: README.md 업데이트 및 새로운 도메인 설명 추가

- Spring 프로젝트 초기화, Bean 선언, Lombok 적용 등 요구사항 상세 업데이트.
- 새로운 도메인, 비즈니스 로직 고도화 및 서비스 설계 추가.
- 체크리스트 세부 내용 보완 및 관련 다이어그램 추가.

commicat2 committed:
feat: application.yml 추가 및 AppProperties 클래스 도입

- application.properties를 application.yml로 변환.
- AppProperties 클래스 생성 및 환경 설정 값 매핑.
- README 프로젝트 초기화 체크리스트 상태 업데이트.
- BCrypt 관련 의존성 추가.

commicat2 committed:
feat: boilerplate

commicat2 committed:
refactor: JCF 기반 서비스 제거 및 관련 로직 정리

- `BaseJcfService` 및 `JcfFriendRequestService` 클래스 삭제.
- `Message` 및 `BaseEntity` 클래스 수정으로 코드 간결화.

commicat2 committed:
feat: `DevUserService` 메소드 리팩토링 및 테스트 환경 설정 추가

- `register` 및 `login` 메소드 리턴 타입을 `UUID`로 변경.
- `BaseEntity` 클래스에서 `equals`, `hashCode` 메소드 수정.
- `test` 프로필 환경 설정 및 애플리케이션 초기화 코드 추가.
- 불필요한 코드 정리 및 리팩토링 진행.

commicat2 committed:
feat: `@ConfigurationProperties` 추가 및 Bean 등록

- `AppMetadataProperties`, `AppStorageProperties`, `AppServerProperties`에 `@ConfigurationProperties` 추가.
- 관련 Bean 등록을 위한 `AppConfig` 업데이트.

commicat2 committed:
feat: JCF 및 파일 기반 메시지 서비스 제거, Dev 환경 서비스 도입

- `JcfMessageService`와 `FileMessageService` 제거.
- `DevMessageService`, `DevChannelService`, `DevChatRoomService`, `DevGuildService` 추가.
- `ApplicationRunner`에 `FriendRequestService` 관련 dev/test 프로필 지원 추가.
- `MessageService`에 `send` 메소드 추가.

commicat2 committed:
feat: DevFriendRequestService 추가 및 기존 FriendRequestService 리팩토링

- `DevFriendRequestService` 인터페이스 및 구현체 추가.
- 기존 FriendRequestService에 리스트 기능(`listSent`, `listReceived`, `listAllMine`) 추가.
- `send`, `reject` 메서드 동작 방식 개선 및 예외 처리 추가.
- Dev 환경 초기화를 위한 `ApplicationRunner`에 관련 로직 통합.

commicat2 committed:
feat: DevUserService 및 관련 클래스 리팩토링

- DevUserService 인터페이스 분리 및 구현 클래스 BasicUserService로 변경.
- register, login 메소드 리턴 타입을 UUID에서 User로 변경.
- FriendRequestService 및 해당 구현체에서 send 메소드 리턴 타입 변경.
- FriendRequestRepository 메서드 리팩토링 및 deleteAllByIds 제거.
- UserService 관련 메서드에서 Set<> 반환 타입을 List<>로 변경.

commicat2 committed:
dev 환경 설정 추가 및 파일 기반 데이터 로직 수정

- application.yml에서 active 프로필 `dev`로 변경.
- `FileBaseRepository` 로직 간소화 및 예외 처리 방식 변경.
- `BasicFriendRequestService`에 dev 프로필 추가.
- 파일 기반 데이터 변경 및 DevUser 관련 파일 갱신.

commicat2 committed:
dev 환경 설정 추가 및 파일 기반 데이터 로직 수정

- application.yml에서 active 프로필 `dev`로 변경.
- `FileBaseRepository` 로직 간소화 및 예외 처리 방식 변경.
- `BasicFriendRequestService`에 dev 프로필 추가.
- 파일 기반 데이터 변경 및 DevUser 관련 파일 갱신.

commicat2 committed:
refactor: `deventity`를 `entitydev`로 패키지 이름 변경

commicat2 committed:
refactor: `JcfBaseRepository` 메서드 로직 간소화 및 코드 정리

- `findAllIncludingDeleted` 메서드에서 `ArrayList`를 `List.copyOf`로 변경.
- `existsAllByIds` 메서드에서 빈 값 확인 로직 수정.
- 불필요한 `import` 제거.

commicat2 committed:
파일 기반 DevUser 및 DevGuild 데이터 추가

commicat2 committed:
refactor: DM 채팅방 참가자 관리 로직 개선 및 관련 서비스 리팩토링

- DM 채팅방의 참가자 추가/제거 시 해시코드 갱신 로직 추가.
- `participantsHashcode` 필드 추가 및 관련 메서드 리팩토링.
- `DevMessageService`와 관련 서비스 메서드 예외 처리 추가.
- JCF 및 파일 기반 저장소 로직 정리 및 DM 전용 유효성 검사 강화.
- `guild`와 `friend request`의 예외 처리 및 정렬 로직 추가.
- null 및 빈 값 검증 로직 개선.

commicat2 committed:
feat: Dev 환경 서비스 인터페이스 추가

- `DevChatRoomService`, `DevGuildService`, `DevChannelService` 인터페이스 정의.
- 각 인터페이스에 메시지, 참가자, 채널, 멤버 관리 메서드 추가.
- Dev 환경의 Guild, Channel, ChatRoom 관리 지원 강화.

commicat2 committed:
feat: JCF 및 파일 기반 채널 서비스 제거, Dev 환경 채널 서비스 도입

commicat2 committed:
feat: JCF 및 파일 기반 서비스 제거, Jpa 및 Dev 환경 서비스 도입

- JCF 및 파일 기반 Guild, ChatRoom, Channel 서비스 제거.
- 새로운 Jpa 채널 및 메시지 서비스 구현체 추가.
- Dev 환경 채널 및 채팅방 서비스 리팩토링 및 관련 저장소 통합.
- 서비스 인터페이스 생성 메서드 및 메시지 반환 구조 변경.

commicat2 committed:
refactor: 서비스 클래스에 `@RequiredArgsConstructor` 적용 및 생성자 제거

- Lombok을 활용하여 각 서비스 클래스의 생성자를 `@RequiredArgsConstructor`로 대체.
- 관련 클래스의 필드 초기화 자동화.
- 불필요한 코드 간소화 및 가독성 향상.

commicat2 committed:
refactor: 이름 변경 및 UserStatus 서비스 추가

commicat2 committed:
feat: BinaryContent 도메인 및 서비스 구현

- BinaryContent 엔티티에 파일 관련 필드 및 유효성 로직 추가.
- BinaryContent 서비스에서 CRUD 메서드 및 DTO 적용.
- JCF 및 파일 기반 저장소 구현체 추가.
- README와 데이터 파일 갱신.

commicat2 committed:
refactor: 서비스 인터페이스 제거 및 구현 클래스 리팩토링

- 불필요한 서비스 인터페이스 삭제.
- Dev 환경 서비스 구현 클래스에서 인터페이스 의존성 제거.
- 각 서비스에서 중복된 `override` 메서드 삭제.
- 일부 도메인 및 DTO 필드 구조 간소화.

commicat2 committed:
refactor: 도메인 및 서비스 필드 네이밍 및 메서드 개선

commicat2 committed:
refactor: 서비스 인터페이스 및 파일 기반 구현 제거, DTO와 서비스 메서드 개선

- 불필요한 서비스 인터페이스 제거 및 관련 구현 클래스 리팩토링.
- Jpa 및 Dev 환경 서비스로 전환.
- DTO 구조 간소화 및 새로운 명령/응답 객체 추가.
- Auth 및 Guild 서비스 메서드 내 데이터 초기화 로직 수정.
- 파일 및 BinaryContent 기반 데이터 제거, 관련 파일 삭제.

commicat2 committed:
style

commicat2 committed:
docs: sprint4 readme 변경

commicat2 committed:
feat: 유저 CRUD web API 구현, 웹 API 예외 전역 처리 추가

commicat2 committed:
feat: BinaryContentController 추가

- 기존 명령(DTO) 객체를 요청(Request) 객체로 변경 및 간소화.
- BinaryContent 관련 서비스 메서드 개선: 파라미터 및 유효성 검증 로직 수정.
- BinaryContentController 추가 및 파일 업로드 엔드포인트 구현.
- 파일 이름 관련 유틸 클래스(FileNames) 도입.

commicat2 committed:
feat: 프로필 이미지 업데이트 기능 추가 및 관련 로직 개선

- `UserUpdateProfileImageRequest` 추가 및 기존 데이터 구조 수정.
- `BinaryContentService`를 활용한 프로필 이미지 조회 및 업데이트 로직 구현.
- `UserController`에 프로필 이미지 업데이트 엔드포인트 추가.

commicat2 committed:
feat: BinaryContentResponse에 size 필드 추가

- BinaryContentService에서 BinaryContentResponse의 size 매핑 로직 추가.

commicat2 committed:
feat: AuthController 및 BinaryContent API 개선

- AuthController 추가: 로그인/로그아웃 엔드포인트 구현.
- BinaryContentService에 findAll 메서드 추가, 관련 API 수정.
- UserController에서 이메일 업데이트 엔드포인트 추가 및 관련 DTO 수정.
- 도메인, 서비스, DTO, 컨트롤러 네이밍 및 구조 개선.
- README 업데이트: API 테스트 및 상태 반영.

commicat2 committed:
데이터 파일 추가: User, UserStatus, BinaryContent 신규 파일 생성

commicat2 committed:
feat: 유저 및 상태 관리 로직 개선, DTO 및 컨트롤러 업데이트

- `UserMapper` 추가: User와 Status를 UserResponse로 변환.
- `FriendController` 추가: 친구 관련 컨트롤러 생성.
- DTO 리팩토링: Command 객체 -> Request 객체로 변경 및 간소화.
- `UserStatusRepository`에 batch 조회 메서드 추가.
- `UserStatusService` 및 `BasicUserService` 로직 리팩토링.
- `UserController`에 사용자명, 비밀번호 업데이트 및 계정 비활성화 API 추가.
- 불필요한 코드 및 예전 로직 제거.

commicat2 committed:
feat: 친구 요청 및 처리 로직 개선, FriendController 삭제

- `FriendController` 제거 및 관련 로직 `UserController`로 통합.
- 새로운 DTO 추가 (`FriendRequestSendRequest`, `FriendRequestHandleRequest`, `FriendRequestResponse`).
- `FriendRequestMapper` 추가: 요청 데이터를 응답으로 매핑.
- `BasicFriendRequestService` 로직 확장: 친구 요청 전송, 조회, 처리 기능 포함.
- 친구 관련 API (`getFriends`, `removeFriend`, `getFriendRequests`, `sendFriendRequest`, `handleFriendRequest`) 추가.

commicat2 committed:
fix: api endpoint 수정

commicat2 committed:
feat: API 개선 및 유효성 검증 로직 수정

- 친구 요청 및 처리 로직 FriendRequestController로 분리.
- BinaryContent 관련 엔드포인트 및 데이터 필드 수정.
- 전역 예외 처리에 ValidationException 추가.
- 유효성 검증 로직 확장(Jakarta Validation 적용).
- AppProperties에 NestedConfigurationProperty 적용.
- 불필요한 설정 및 코드 정리.
- 로깅 레벨 조정 및 디버깅 개선.
- 새로운 의존성 추가(Spring Boot Validation).

commicat2 committed:
feat: BinaryContent 및 BaseEntity 리팩토링

- BinaryContent: 메타데이터 항목 추가(sha256, storagePath 등), 유효성 검증 및 static 유틸 메서드 개선.
- BaseEntity: soft delete 및 버전 관리 기능 추가, 관련 메서드 확장.
- 코멘트 추가 및 코드 구조 개선.

commicat2 committed:
refactor: ChatRoom 및 관련 서비스/레포지토리 제거

- ChatRoom 엔티티와 관련 서비스(BasicChatRoomService, JpaChatRoomService) 삭제.
- 관련 레포지토리와 메서드 제거(FileChatRoomRepository, JcfChatRoomRepository 등).
- 사용되지 않는 클래스(RoleType, JpaChannelService 등) 삭제.
- 코드 및 구조 단순화.

commicat2 committed:
feat: NotFoundException 및 AccessDeniedException 추가, 예외 처리 개선

- NotFoundException, AccessDeniedException 클래스 생성.
- 유저 조회 및 삭제 로직에서 NotFoundException 활용.
- GlobalExceptionHandler에 예외 처리 추가 (NotFoundException, AccessDeniedException, 기타 예외 처리 개선).

azjaehyun reviewed (approved):
지석님 코드 리펙토링 하고 계신거 같은데 리팩토링이 다 끝나고 코드리뷰를

해드려야될것 같습니다.

꼭 한번 모범 답안 코드를 보고 개발을 진행하셨으면 좋겠습니다.

jpa 를 아직 배우지도 않았는데 jpa 코드가 있으며.

요구사항에 필요한 파일은 있지 않고 다른 코드들이 너무 많은거 같아요.

수업듣고 일도 하시느라 많이 바쁘시고 시간 내기 힘든거 알지만

꼭 한번 모범답안 코드를 보시고 그거 기반으로 진행하셨으면 좋겠습니다.

commicat2 committed:
refactor: 컨트롤러 및 DTO 리팩토링, 유효성 검증 로직 개선

- @**Mapping으로 메서드 매핑 정리 및 불필요한 RequestMethod 제거.
- 각 DTO에 Jakarta Validation 추가(@NotNull, @NotBlank, @Email 등 적용).
- 컨트롤러 및 서비스 메서드 로직 간소화 및 표현식 개선.
- BinaryContentMapper와 UserMapper 등 새로운 매퍼 클래스 도입.
- FriendRequest 및 관련 엔드포인트 리팩토링.

commicat2 commented:
sprint4 브랜치에서 작업을 하는 중이어서 일단 sprint3으로 작업하던 내용을 머지해서 푸시했습니다.
스케줄 관리를 잘못해서 요구사항대로 완성하지 못했었습니다. 죄송합니다.
리뷰하기 어려우시면 바로 merge해주셔도 괜찮습니다.

commicat2 committed:
fix: 서비스 클래스에 @Transactional 추가 및 일부 데이터 파일 갱신

commicat2 committed:
feat: UserMapper 및 UserResponse에 bio 필드 추가

- UserMapper 매핑 로직에 bio 필드 추가.
- UserResponse DTO에 bio 항목 반영.

commicat2 committed:
fix: User 엔티티의 setGlobalName 버그 수정

commicat2 committed:
refactor: BinaryContentController 메서드 매핑 간소화 및 일부 데이터 파일 갱신

- @**Mapping으로 메서드 매핑 변경(RequestMethod 제거).
- 불필요한 공백 제거 및 코드 정리.
- 관련 바이너리 데이터 파일 갱신.

commicat2 committed:
refactor: FriendRequestController 및 BinaryContentController 메서드 매핑 변경

- @**Mapping으로 메서드 매핑 수정(RequestMethod 제거).
- 코드 간소화 및 불필요한 import 정리.

commicat2 committed:
refactor: BinaryContent 관련 필드명 및 메서드명 변경

- `data`를 `bytes`로 필드 및 메서드명 수정.
- 관련 엔티티, DTO, Controller 및 Service 전체 변경 및 코드 정리.
- README 예제 수정 및 문장 간소화.
